### PR TITLE
Add Versioning

### LINKED ISSUE #2247: Versioning of Docs
### Describe your feature

While evolving Gitpod Self-Hosted we need to make sure the documentation is versioned, so people who are on an older Gitpod release are able to find their relevant information.

### User impact

Users who are running an older release, can look into older documentation, which may decrease the amount of support requests

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Cahllagerfeld**
closes #2247

https://deploy-preview-2725--gitpod-kumquat.netlify.app/

<a href="https://gitpod-staging.com/#https://github.com/gitpod-io/website/pull/2725"></a>

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/2725"></a>

### CODE REVIEW

**Comment by Siddhant-K-code** [comment]
This PR is so huge. I'm just worried about the Review & potential breaking changes.
Also, there will be a lot of conflict due to #2587

When are you planning to Ship this?

**Comment by Cahllagerfeld** [comment]
Planned early next week

**Comment by Cahllagerfeld** [comment]
Hey @lucasvaltl, 
thats the draft PR for versioning the entire docs. I added a dummy version for testing purposes.
Things I did:
- Make links relative
- removed `latest` tag in self-hosted docs
- make left-side-menu dynamic

Note: The build time increased from about 4 min to 6 min.

‼️ Before merging we need to revert the last commit.

File: gitpod/docs/configure/self-hosted/latest/reference-architecture/_chunks/cluster.md [hunk 1]
```diff
@@ -6,21 +6,17 @@ layout: false
   import CloudPlatformToggle from "$lib/components/docs/cloud-platform-toggle.svelte";
```
Comment (Cahllagerfeld) [inline gitpod/docs/configure/self-hosted/latest/reference-architecture/_chunks/cluster.md:6]: Double check these files

**Comment by Cahllagerfeld** [comment]
closing this issue for now, because it's on hold for a long time now

**Pull request closed without merging**